File: backend/users-ms/src/main/java/com/azat4dev/demobooking/users/users_queries/domain/entities/PersonalUserInfo.java
package com.azat4dev.demobooking.users.users_queries.domain.entities;

import com.azat4dev.demobooking.users.common.domain.values.UserId;
import com.azat4dev.demobooking.users.users_commands.domain.core.values.user.FullName;

import java.util.Optional;


public record PersonalUserInfo(
    UserId id,
    String email,
    FullName fullName,
    Optional<UserPhoto> photo
) {
}

File: web-client/src/domain/auth/values/PersonalUserInfo.ts
import UserId from "./UserId";
import Email from "./Email";
import FullName from "./FullName";
import {PhotoPath} from "./PhotoPath";

class PersonalUserInfo {
    public constructor(
        public readonly id: UserId,
        public readonly email: Email,
        public readonly fullName: FullName,
        public readonly photo: PhotoPath | null
    ) {
    }
}

export default PersonalUserInfo;